exasol_integration_test_docker_environment/test/test_api_test_environment.py

```diff
@@ -77,6 +81,77 @@ def create_db_connection_command(self):
         bash_cmd = f"""bash -c "{cmd}" """
         return bash_cmd
 
+    def test_build_mapping_working(self):
+        environment_info = self.environment.environment_info
+        with ContextDockerClient() as docker_client:
+            test_container = docker_client.containers.get(environment_info.test_container_info.container_name)
+            exit_code, output = test_container.exec_run("cat /test.text")
+            self.assertEqual(exit_code, 0)
+            self.assertEqual(output.decode("UTF-8"), "Empty File")
+
+
+class APISpawnTestEnvironmentTestWithCustomRuntimeMapping(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        print(f"SetUp {cls.__name__}", file=stderr)
+        cls.test_environment = ApiTestEnvironment(cls)
+        cls.docker_environment_name = cls.__name__
+
+    @classmethod
+    def tearDownClass(cls):
+        utils.close_environments(cls.test_environment)
+
+    def test_runtime_mapping_without_deployment(self):
+        with tempfile.TemporaryDirectory() as temp_dir:
+            temp_path = Path(temp_dir)
+            with open(temp_path / "test.txt", "w") as f:
+                f.write("test")
+            environment = None
+            mapping = TestContainerRuntimeMapping(source=temp_path, target="/test")
+            try:
+                environment = \
+                    self.test_environment.spawn_docker_test_environment_with_test_container(
+                        name=self.docker_environment_name,
+                        test_container_content=get_test_container_content((mapping,))
+                    )
+                environment_info = environment.environment_info
+                with ContextDockerClient() as docker_client:
+                    test_container = docker_client.containers.get(environment_info.test_container_info.container_name)
+                    exit_result = test_container.exec_run("cat /test/test.txt")
+                    exit_code = exit_result[0]
+                    output = exit_result[1]
+                    self.assertEqual(exit_code, 0)
+                    self.assertEqual(output.decode("utf-8"), "test")
```

Comment:
extract to function deployment_available

Reply:
Done